### PR TITLE
SUS-4790 / SUS-4791 | respect wgEnableHAWelcomeExt value coming from WikiFactory

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-4790 and https://wikia-inc.atlassian.net/browse/SUS-4791

And a fix a notice from a previous fix - #15326

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: includes/wikia/Extensions.php [hunk 1]
```diff
@@ -705,7 +705,7 @@
 }
 
 // Enable CategorySelect extension for all not RTL wikis
-if (!in_array($wgLanguageCode, array('ar', 'fa', 'he', 'ps', 'yi')) && is_null( $wgEnableCategorySelectExt) ) {
+if (!in_array($wgLanguageCode, array('ar', 'fa', 'he', 'ps', 'yi')) && !isset($wgEnableCategorySelectExt) ) {
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:708]: I believe this won't work if `wgEnableCategorySelectExt` is explicitly set to `false` in WikiFactory.

**Review by mszabo-wikia — DISMISSED** [review]

File: includes/wikia/Extensions.php (hunk 1 above)
Comment (macbre) [reply includes/wikia/Extensions.php:708]:
```php
> $a=false;

> var_dump( isset($a))
bool(true)
```

Comment (ghost) [reply includes/wikia/Extensions.php:708]: Gotta love PHP. So `empty()` perhaps?

Comment (mszabo-wikia) [reply includes/wikia/Extensions.php:708]: My bad, I misremembered—this happens when you set an array entry to `null`, then it'll be considered "not set".

**Pull request merged**